File: src/main/java/com/learning/java/controlstruct/LoopsBreaksSyntax.java
package com.learning.java.controlstruct;

import java.util.Arrays;
import java.util.List;

/* Java: cинтаксис цыклов и управления прохождения по цыклу */


public class LoopsBreaksSyntax {

	public static void main(String[] args) {
        System.out.println("################");
		/* (#1)  while */
		int i = 0;
		while(i < 3) {
			System.out.println(i);
			i++;
		}

        System.out.println("\n################");
		/* (#2.1)  for */
		for(int j = 0; j < i; j++) {
			System.out.println(j);
		}

        System.out.println("\n################");
		/* (#2.2)  for:each */
		List<Integer> numbers = Arrays.asList(0, 1, 2, 3);
		numbers.forEach(n -> System.out.println(n)); /* альтарнативный синтаксис в Java-8:  numbers.forEach(System.out::println); */

        System.out.println("\n################");
		/* (#3)  for:break */
		for(int k = 0; k < 10; k++) {
			System.out.println(k);
			if(k == 5)
				break;			
		}
	}
}

File: src/main/scala/com/learning/scala/controlstruct/LoopsBreaksSyntax.scala
package com.learning.scala.controlstruct

/* Scala: cинтаксис цыклов и управления прохождения по цыклу */


object LoopsBreaksSyntax {

  def main(args: Array[String]) {
    println("################")
    /* (#1)  while */
    var i: Int = 0
    while (i < 3) {
      println(i)
      i += 1 // ( в Scala отсутствует оператор: ++ )
    }

    println("\n################")
    /* (#2.1)  for */
    for(i <- 0 to i) { /* <-  указывает на генератор чтобы увеличить значение в итераторе */
      println(i)
    }

    println("\n################")
    /* (#2.2)  for:each */
    (0 to 3).foreach(i => println(i)) /* альтернативный синтаксис использования в Scala:  (0 to 3).foreach(println(_) */

    println("\n################")
    /* (#3)  for:break */
    import scala.util.control.Breaks._ // ( также Scala позволяет делать import внутри метода или блока )
    breakable {
      for(k <- 0 to 10) {
        println(k)
        if(i ==5)
          break()
      }
    }
  }
}
